src/Resources/config/twig_string.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Sonata\AdminBundle\Twig\Extension\DeprecatedTextExtension;
+use Sonata\AdminBundle\Twig\Extension\StringExtension;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $containerConfigurator->services()
+
+        ->set('sonata.string.twig.extension', StringExtension::class)
+            ->tag('twig.extension')
+            ->args([
+                null,
+            ])
+
+        ->set('sonata.deprecated_text.twig.extension', DeprecatedTextExtension::class)
+            ->tag('twig.extension')
+            ->deprecate('The "%service_id%" service is deprecated. You should stop using it, as it will be removed in the future.')
```

Comment:
```suggestion
            ->deprecate('The "%service_id%" service is deprecated since sonata-project/admin-bundle 3.x. You should stop using it, as it will be removed in 4.0.')
```
We should find which version deprecated this service in order to make this message explicit.
Even, we should also add a `NEXT_MAJOR` comment explaining what we must do about this in the next major release.

Reply:
I added 3.70, which is where https://github.com/sonata-project/SonataAdminBundle/pull/6145 was included.